lib/imgix/rails/tag.rb

```diff
@@ -31,127 +31,43 @@ def srcset(url_params: @url_params, widths: @widths)
 
 private
 
+  MINIMUM_SCREEN_WIDTH = 100
   MAXIMUM_SCREEN_WIDTH = 2560 * 2 # Physical resolution of 27" iMac (2016)
-  SCREEN_STEP = 100
-
-  # Taken from http://mydevice.io/devices/
-
-  # Phones
-  IPHONE = { css_width: 320, dpr: 1 }
-  IPHONE_4 = { css_width: 320, dpr: 2 }
-  IPHONE_6 = { css_width: 375, dpr: 2 }
-  LG_G3 = { css_width: 360, dpr: 4 }
-
-  # Phablets
-  IPHONE_6_PLUS = { css_width: 414, dpr: 3 }
-  IPHONE_6_PLUS_LANDSCAPE = { css_width: 736, dpr: 3 }
-  MOTO_NEXUS_6 = { css_width: 412, dpr: 3.5 }
-  MOTO_NEXUS_6_LANDSCAPE = { css_width: 690, dpr: 3.5 }
-  LUMIA_1520 = { css_width: 432, dpr: 2.5 }
-  LUMIA_1520_LANDSCAPE = { css_width: 768, dpr: 2.5 }
-  GALAXY_NOTE_3 = { css_width: 360, dpr: 3 }
-  GALAXY_NOTE_3_LANDSCAPE = { css_width: 640, dpr: 3 }
-  GALAXY_NOTE_4 = { css_width: 360, dpr: 4 }
-  GALAXY_NOTE_4_LANDSCAPE = { css_width: 640, dpr: 4 }
-
-  # Tablets
-  IPAD = { css_width: 768, dpr: 1 };
-  IPAD_LANDSCAPE = { css_width: 1024, dpr: 1 };
-  IPAD_3 = { css_width: 768, dpr: 2 };
-  IPAD_3_LANDSCAPE = { css_width: 1024, dpr: 2 };
-  IPAD_PRO = { css_width: 1024, dpr: 2 };
-  IPAD_PRO_LANDSCAPE = { css_width: 1366, dpr: 2 };
-
-  BOOTSTRAP_SM = { css_width: 576, dpr: 1 }
-  BOOTSTRAP_MD = { css_width: 720, dpr: 1 }
-  BOOTSTRAP_LG = { css_width: 940, dpr: 1 }
-  BOOTSTRAP_XL = { css_width: 1140, dpr: 1 }
-
-  def devices
-    phones + phablets + tablets + bootstrap_breaks
-  end
-
-  def bootstrap_breaks
-    breaks = [
-      BOOTSTRAP_SM,
-      BOOTSTRAP_MD,
-      BOOTSTRAP_LG,
-      BOOTSTRAP_XL
-    ]
+  INCREMENT_PRECENTAGE = 8
 
-    breaks + breaks.map { |b| b[:dpr] = 2; b }
-  end
+  @@standard_widths = nil
 
-  def phones
-    [
-      IPHONE,
-      IPHONE_4,
-      IPHONE_6,
-      LG_G3
-    ]
-  end
+  def standard_widths
+    return @@standard_widths if @@standard_widths
 
-  def phablets
-    [
-      IPHONE_6_PLUS,
-      IPHONE_6_PLUS_LANDSCAPE,
-      MOTO_NEXUS_6,
-      MOTO_NEXUS_6_LANDSCAPE,
-      LUMIA_1520,
-      LUMIA_1520_LANDSCAPE,
-      GALAXY_NOTE_3,
-      GALAXY_NOTE_3_LANDSCAPE,
-      GALAXY_NOTE_4,
-      GALAXY_NOTE_4_LANDSCAPE
-    ]
-  end
+    prev = MINIMUM_SCREEN_WIDTH
+    @@standard_widths = []
+    while prev < MAXIMUM_SCREEN_WIDTH do
+      @@standard_widths.append(2 * (prev/2).round)   # Ensure widths are even
+      prev = prev * (1 + (INCREMENT_PRECENTAGE/100.0)*2.0)
```

Comment:
The user should be able to override the percentage used, here. I'm not sure on the option name, though. Maybe `srcset_tolerance`?

Reply:
Also, should this value be provided as a integer between 1 and 100, or a unit scalar between 0 and 1? I'd opt for the latter, personally, but curious what others think.

Reply:
> The user should be able to override the percentage used, here. I'm not sure on the option name, though. Maybe `srcset_tolerance`?

`srcset_tolerance` sounds good to me.

> Also, should this value be provided as a integer between 1 and 100, or a unit scalar between 0 and 1? I'd opt for the latter, personally, but curious what others think.

I went with 1 to 100 because I thought we'd agreed on it.

I prefer 0 to 1 as well because it implies it's a multiplier. With 1 to 100, we need to indicate it's a percentage value.

Reply:
Let's go with a unit-scalar, then. Obviously the best values for this option lie somewhere between `0.05` and `0.50`, but the reasonable range should be outlined in the documentation rather than enforced in code (except for enforcing that the value must be greater than zero, obviously).

Reply:
To me `tolerance` indicates a threshold or other value that needs to be exceeded. Does it have the same meaning for you guys? For me something like `srcset_scale_rate` makes more sense, or even `srcset_max_diff_from_orig`.

Happy to go either way with the unit scalar. I'm wondering if it would be better to ask people to provide values greater than 1, to make it more clear what is going on? To me, if I provide a value between 0 and 1, I'm expecting it to decrease by that much every time... if you folk disagree, that's all good.